### ISSUE TITLE
Bug: Workflow stuck at the Annotating resources step

### ISSUE THREAD

**Issue by lucafaggianelli** [Azure/k8s-deploy#267]
### What happened?

The k8s-deploy action get stuck at the `Annotating resources` step:

![image](https://user-images.githubusercontent.com/4471043/205890137-9982231e-281b-4c72-aae0-9143c99f8ee4.png)

Everything goes fine and the pods are correctly deployed, but the Github Actions job get stuck and I need it to cancel it manually or wait for it to timeout, resulting in a failed run.

I tried also to set the option `annotate-namespace` to false but nothing changes

### Version

- [X] I am using the latest version

### Runner

self-hosted

### Relevant log output

Debug log of the annotating resources step:

```
Annotating resources
  ##[debug]Kubectl run with command: /opt/hostedtoolcache/kubectl/1.25.4/x64/kubectl get,pods,-o,json,--namespace,engie-dash
  ##[debug]Getting workflows for repo: gempy/engie-dash
  ##[debug]Re-evaluate condition on job cancellation for step: 'Deploy to Kubernetes'.
  Error: The operation was canceled.
  ##[debug]System.OperationCanceledException: The operation was canceled.
  ##[debug]   at System.Threading.CancellationToken.ThrowOperationCanceledException()
  ##[debug]   at GitHub.Runner.Sdk.ProcessInvoker.ExecuteAsync(String workingDirectory, String fileName, String arguments, IDictionary`2 environment, Boolean requireExitCodeZero, Encoding outputEncoding, Boolean killProcessOnCancel, Channel`1 redirectStandardIn, Boolean inheritConsoleHandler, Boolean keepStandardInOpen, Boolean highPriorityProcess, CancellationToken cancellationToken)
  ##[debug]   at GitHub.Runner.Common.ProcessInvokerWrapper.ExecuteAsync(String workingDirectory, String fileName, String arguments, IDictionary`2 environment, Boolean requireExitCodeZero, Encoding outputEncoding, Boolean killProcessOnCancel, Channel`1 redirectStandardIn, Boolean inheritConsoleHandler, Boolean keepStandardInOpen, Boolean highPriorityProcess, CancellationToken cancellationToken)
  ##[debug]   at GitHub.Runner.Worker.Handlers.DefaultStepHost.ExecuteAsync(IExecutionContext context, String workingDirectory, String fileName, String arguments, IDictionary`2 environment, Boolean requireExitCodeZero, Encoding outputEncoding, Boolean killProcessOnCancel, Boolean inheritConsoleHandler, String standardInInput, CancellationToken cancellationToken)
  ##[debug]   at GitHub.Runner.Worker.Handlers.NodeScriptActionHandler.RunAsync(ActionRunStage stage)
  ##[debug]   at GitHub.Runner.Worker.ActionRunner.RunAsync()
  ##[debug]   at GitHub.Runner.Worker.StepsRunner.RunStepAsync(IStep step, CancellationToken jobCancellationToken)
  ##[debug]Finishing: Deploy to Kubernetes
```

**Comment by jaiveerk** [comment]
Hey @lucafaggianelli,
Thanks for raising this issue! Sorry that this has been causing you trouble. 

We believe this may be related to #264 which we've already created a fix for, and it will be part of our next release.

**Comment by lucafaggianelli** [comment]
cool, I'll try the next release thanks!

**Comment by jaiveerk** [comment]
Working on getting one out now! Will update as soon as it's out

**Comment by jaiveerk** [comment]
Release should be out with 4.8!

**Comment by lucafaggianelli** [comment]
thanks for the release but sadly I have a new error (I'm on GH Enterprise):

![image](https://user-images.githubusercontent.com/4471043/206136501-22fcfabd-bd12-4c73-8ab5-7e09de843498.png)

**Comment by lucafaggianelli** [comment]
@jaiveerk should I open another issue?

**Comment by jaiveerk** [comment]
Hm, that's interesting - not sure how something on our end could cause an error there. I wonder if an intermediate step is failing, would you mind running with debug logs?

Correct me if I'm wrong or misunderstanding, but the timeout message you posted above looks like the action timed out because our step was taking too long, but it looks like here the runner is failing to pull information from your workflows?

**Comment by jaiveerk** [comment]
Has this action worked for you in the past? I wonder if the issue is stemming from restricted repo access.

**Comment by lucafaggianelli** [comment]
as far as I tried this action, it never worked for me, v4.7 hanged forever at the Annotating resources step (this issue) and now this error. BTW if I remove the k8s-deploy action, the workflow passes

**Comment by jaiveerk** [comment]
We believe this may be due to an error with the runner's ability to access your repo due to the fact that you appear to be using a custom-hosted Github repo. 

To resolve the issue for now, we've added an extra catch so that your action doesn't fail as a result of one single failed annotation attempt.

**Comment by jaiveerk** [comment]
Sounds good! I'm waiting on PR approval from my team and we need to cut a new release, but I'll let you know when that's done!

**Comment by OliverMKing** [comment]
Reopening to track this until we release.
